examples/places-utils/src/places-utils.rs

```diff
@@ -366,9 +204,18 @@ fn main() -> Result<()> {
 
     let db_path = opts.database_path;
     let api = PlacesApi::new(&db_path)?;
-    let db = api.open_connection(ConnectionType::ReadWrite)?;
     // Needed to make the get_registered_sync_engine() calls work.
     api.clone().register_with_sync_manager();
+    ctrlc::set_handler(move || {
+        if !shutdown::in_shutdown() {
+            println!("\nCTRL-C detected, enabling shutdown mode\n");
+            shutdown::shutdown();
+        } else {
+            println!("\nCTRL-C detected disabling shutdown mode\n");
```

Comment:
this seems a little suspect - if I hit ctrl+c multiple times I might end up in a frankenstate? (Not that I really care for this utility, but I'm trying to understand things)